### ISSUE TITLE
`restartApp()` does not clean up correctly

### ISSUE THREAD

**Issue by tomgroenwoldt** [gabrimatic/restart_app#38]
Hey :)

I have an application that is connected to another device using Android Open Accessory. The communication is handled part-wise in kotlin and part-wise in dart. When I call the `restartApp()` function, the app is restarted but the communication between my devices is not cleaned up correctly. This results in an error on startup when trying to reconnect to my device.

I tested the older version `1.2.1` where the `exit()` function was called internally. This version solved my problem. I'm happy to be part of testing a solution! :)

**Comment by vinz-mehra** [comment]
Yes, using it with Isar in my project. It was working fine on 1.2.1 but started giving errors on 1.3.2

**Comment by augustodia** [comment]
I'm using Shorebird for Codepush. One of the requirements to be able to apply the update is to restart the app. On iOS, restarting actually applies the updates. On Android, restarting isn't as thorough apparently, as I need to manually close the app and open it in order to apply the updates.

**Comment by DHT-Xavier** [comment]
> I'm using Shorebird for Codepush. One of the requirements to be able to apply the update is to restart the app. On iOS, restarting actually applies the updates. On Android, restarting isn't as thorough apparently, as I need to manually close the app and open it in order to apply the updates.

Same usage here and it worked before. Now even called `Restart.restartApp();` the app seems restarted, however Shorebird still showing `restartRequired` until user manually close the app and launch again.

Looking forward to see an update to fix this issue

**Comment by xalikoutis** [comment]
Here the same also with shorebird user has to terminate the app in order to start with patch version

**Comment by leandro-pilz** [comment]
The same thing happens here
